.github/workflows/pull-noir.yml

```diff
@@ -65,85 +63,35 @@ jobs:
-            STAGING_BRANCH=$BRANCH
-          else
-            # otherwise we first reset our staging branch
-            STAGING_BRANCH=$BRANCH-staging
-          fi
-
-          # Get the last sync PR's last commit state
-          COMMIT_MESSAGE=$(cat .PR_BODY_MESSAGE)
-          LINES=$(echo $COMMIT_MESSAGE | wc -l)
-
-          function force_sync_staging() {
-            # reset to last noir merge
-            git checkout $STAGING_BRANCH || git checkout -b $STAGING_BRANCH
-            git reset --hard "$BASE_AZTEC_COMMIT"
-            # Reset our branch to our expected target
-            git push origin $STAGING_BRANCH --force
-            # force gitrepo to point to the right HEAD (we ignore .gitrepo contents otherwise)
-            git config --file="$SUBREPO_PATH/.gitrepo" subrepo.commit "$BASE_NOIR_COMMIT"
-            # we need to commit for git-subrepo
-            git commit -am "[$LINES changes] $COMMIT_MESSAGE"
-            if ./scripts/git-subrepo/lib/git-subrepo pull --force $SUBREPO_PATH --branch=master; then
-              # Read our actual commit sync from git subrepo, stash to file for next time
-              COMMIT=$(git config --file="$SUBREPO_PATH/.gitrepo" subrepo.commit)
-              echo "$COMMIT" > .noir-sync-commit && git add .noir-sync-commit
-              git reset --soft "$BASE_AZTEC_COMMIT"
-              # We don't really need the sync commit on our side, and don't need .gitrepo at all except just in time for the command.
-              git checkout origin/master -- noir/noir-repo/.aztec-sync-commit noir/noir-repo/.gitrepo
-              git commit -am "[$LINES changes] $COMMIT_MESSAGE"
-
-              # There's various changes which we need to make to account for CI differences so we run this script to apply them.
-              git checkout origin/master -- noir/scripts/sync-in-fixup.sh
-              noir/scripts/sync-in-fixup.sh
-              git commit -am "chore: apply sync fixes"
-
-              git push origin $STAGING_BRANCH --force
-            else
-              echo "Problems syncing noir. Needs manual attention, might be a merge conflict."
-              exit 1
-            fi
-          }
-          # merge_staging_branch: Merge our staging branch into aztec-packages.
-          function merge_staging_branch() {
-            # Fix PR branch
-            git fetch # see recent change
-            git checkout $BRANCH || git checkout -b $BRANCH
-            if ! git merge -Xtheirs origin/$STAGING_BRANCH -m "$COMMIT_MESSAGE" ; then
-              # resolve modify/delete conflicts
-              git diff --name-only --diff-filter=U | xargs git rm
-              git commit -m "$COMMIT_MESSAGE"
-            fi
-            git push origin $BRANCH
-          }
-          force_sync_staging
-          if [[ "$PR_URL" != "" ]]; then
-            merge_staging_branch
-          fi
+          BRANCH=bump-noir
+          git fetch --depth 1 origin $BRANCH || true
```

Comment:
It would be nice if this was just calls to noir/bootstrap.sh (github runners will be able to run our scripts just fine as long as they don't rely on anything being built). If we're revisiting this it'd be good to also make that step as it means it's more testable locally. Bonus points if it respects DRY_RUN=1 for just poking out the state without the push step.

Reply:
We can save this until the next time this logic is iterated on, though

Reply:
Okay, I see we have the `do_or_dryrun` to facilitate this. 

I put it in the workflow because that's how the other similar jobs are implemented (e.g. the bumping jobs on Noir side), so I thought it's how it was expected, but I agree that putting them in the script is easier to test. 

I thought about adding the update part to the script as well, but decided against it because doing this whole thing about updating to nightly seems like just one of the many ways we could manage updates, and because we're just reading/writing to that text file; although it did feel like reading and writing should be encapsulated in the script, I wasn't sure if you'd say it's over engineered :)

Reply:
Done